src/ballet/sha256/fd_sha256.h

```diff
@@ -118,6 +125,47 @@ void *
 fd_sha256_fini( fd_sha256_t * sha,
                 void *        hash );
 
+// Advanced interface
+// ==================
+
+/* FD_SHA256_INIT_n are the eight 32-bit integers that comprise the initial hash state.
+   Note that the byte order of each integer is reversed (little-endian). */
+
+#define FD_SHA256_INIT_0 (0x6a09e667U)
+#define FD_SHA256_INIT_1 (0xbb67ae85U)
+#define FD_SHA256_INIT_2 (0x3c6ef372U)
+#define FD_SHA256_INIT_3 (0xa54ff53aU)
+#define FD_SHA256_INIT_4 (0x510e527fU)
+#define FD_SHA256_INIT_5 (0x9b05688cU)
+#define FD_SHA256_INIT_6 (0x1f83d9abU)
+#define FD_SHA256_INIT_7 (0x5be0cd19U)
+
+/* fd_sha256_init_state loads the 32 byte initial state value into the given buffer. */
+
+static inline void fd_sha256_init_state( uchar * state ) {
+  uint * s = (uint *)state;
+  s[0] = FD_SHA256_INIT_0;
+  s[1] = FD_SHA256_INIT_1;
+  s[2] = FD_SHA256_INIT_2;
+  s[3] = FD_SHA256_INIT_3;
+  s[4] = FD_SHA256_INIT_4;
+  s[5] = FD_SHA256_INIT_5;
+  s[6] = FD_SHA256_INIT_6;
+  s[7] = FD_SHA256_INIT_7;
+}
+
+/* fd_sha256_core invokes the internal block function of SHA-256 block_cnt times.
+   For each iteration, reads and advances the block ptr by one block and updates state.
```

Comment:
```
/* fd_sha256_core hashes the block_cnt blocks of data starting at block, updating 
the internal hash state stored in state.  The computation is essentially an 
optimized version of fd_sha256_append( .., block, block_cnt * FD_SHA256_BLOCK_SZ) 
since it does not need to consider padding or buffered data and does not keep track 
of how much data has been hashed.  Additionally, since the caller provides the 
buffer, it can eliminate a memcpy of the final hash.  Because this function does 
not keep track of how much data has been hashed, it's the calling code's 
responsibility to write the equivalent of `fd_sha256_fini`.  This can still be 
helpful if the size is known statically because then the trailing padding can also 
be generated statically.
```